modules/effects/src/main/scala/com/github/gvolpe/fs2redis/transactions.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018-2019 Gabriel Volpe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.github.gvolpe.fs2redis
+
+import cats.Applicative
+import cats.effect._
+import cats.implicits._
+import com.github.gvolpe.fs2redis.algebra._
+import com.github.gvolpe.fs2redis.effect.Log
+
+object transactions {
+
+  private def release[F[_]: Applicative: Log, K, V](
+      cmd: RedisCommands[F, K, V]
+  ): (Unit, ExitCase[Throwable]) => F[Unit] = {
+    case (_, ExitCase.Completed) => cmd.exec *> Log[F].info("Transaction completed")
+    case (_, ExitCase.Error(e))  => cmd.discard *> Log[F].error(s"Transaction failed: ${e.getMessage}")
+    case (_, ExitCase.Canceled)  => cmd.discard *> Log[F].error("Transaction canceled")
+  }
+
+  object Transaction {
+    def apply[F[_]: Log: Sync, K, V, A](cmd: RedisCommands[F, K, V])(fa: F[A]): F[A] =
+      resource(cmd).use(_ => fa)
+
+    def resource[F[_]: Log: Sync, K, V](cmd: RedisCommands[F, K, V]): Resource[F, Unit] =
```

Comment:
I agree it's weird and it's because there's no resource. Maybe the API shouldn't expose a `resource` method? The `apply` should be more than enough and can be implemented using `bracketCase` directly. WDYT?

The only benefit of having it as a resource is that it can be composed but it might not be the way the user wants to use it.